### PR TITLE
Boost: Update query params

### PR DESCRIPTION
**Pull request by haqadn**
## Proposed changes:

* Whitelist `jb-disable-modules` for SEO plugins.
* Some small refactor.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
p1711350424667999/1711195573.320329-slack-C016BBAFHHS

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Test Cloud CSS and local Critical CSS with Yoast and All in One SEO plugin and enable the query parameter removing features. Make sure both C.CSS generation work.

### CODE REVIEW

File: projects/plugins/boost/app/lib/critical-css/Generator.php [hunk 1]
```diff
@@ -63,21 +63,4 @@ public function display_generate_meta() {
...
-
-	/**
-	 * Add the critical css generation flag to a list if it's present in the URL.
-	 * This is mainly used by filters for compatibility.
-	 *
-	 * @var $query_args    array The list to add the arg to.
-	 *
-	 * @return $query_args array The updatest list with query args.
-	 */
-	public static function add_generate_query_action_to_list( $query_args ) {
```
Comment (thingalon) [inline projects/plugins/boost/app/lib/critical-css/Generator.php:75]: This is still [referenced in `Critical_CSS`](https://github.com/Automattic/jetpack/blob/1d819a29859e18bc7ae444361d2e6f35e3c177b7/projects/plugins/boost/app/modules/optimizations/critical-css/Critical_CSS.php#L51).

**Review by thingalon — CHANGES_REQUESTED** [review]
Looks good for Cloud CSS, but it's left a dangling reference in local critical css.

File: projects/plugins/boost/app/lib/critical-css/Generator.php (hunk 1 above)
Comment (haqadn) [reply projects/plugins/boost/app/lib/critical-css/Generator.php:75]: Ops! Updated.

**Review by thingalon — DISMISSED** [review]

**Comment by haqadn** [comment]
Need another check just for phan @thingalon

**Review by thingalon — APPROVED** [review]
phan-tastic

**Pull request merged**